src/lib/template-instance.ts

```diff
@@ -103,6 +108,18 @@ NodeFilter.SHOW_TEXT */
       }
     };
     _prepareInstance(fragment);
+
+    // Since we cloned in the polyfill case, now force an upgrade
+    if (isCEPolyfill) {
+      document.adoptNode(fragment);
+      customElements.upgrade(fragment);
```

Comment:
We should only do this if it's not connected since otherwise it's duplicate work and crawling trees is costly. I think it's fine to merge this and just put a TODO here for now.

Reply:
Just fixed it here